### PR TITLE
add expiresAt() and expiresIn() public static methods and import used…

### PR DESCRIPTION
**Pull request by pine3ree**
… functions

### CODE REVIEW

File: src/SetCookie.php [hunk 1]
```diff
@@ -73,12 +87,63 @@ public static function thatExpires(
...
+        string $sameSite = ''
     ) : SetCookie {
         $expiresAt = (int) $expiresAt->format('U');
 
         return new static($name, $value, $expiresAt, $path, $domain, $secure, $httpOnly, $sameSite);
     }
 
+    /**
+     * Create a cookie that expires in given seconds or time-interval
+     * @param int|null $expiresIn
```
Comment (hansott) [inline src/SetCookie.php]: If I'm not mistaken, we probably need to change this to `@param int|DateInterval|string $expiresIn`?

File: tests/SetCookieTest.php [hunk 2]
```diff
@@ -89,5 +90,96 @@ public function test_it_converts_to_header_value()
...
+        $cookie = SetCookie::thatExpiresIn('name', 'value', $expireStr);
+        $expected = sprintf('name=value; expires=%s', gmdate(self::$HTTP_DATE_FORMAT, $expireTS));
+        $this->assertEquals($expected, $cookie->toHeaderValue());
+
+        // test 0 string expression interval
+        $expireStr = '0 day';
+        $expireSecs = 0 * 86400;
+        $expireTS = time() + $expireSecs;
+        $cookie = SetCookie::thatExpiresIn('name', 'value', $expireStr);
+        $expected = sprintf('name=value; expires=%s', gmdate(self::$HTTP_DATE_FORMAT, $expireTS));
```
Comment (hansott) [inline tests/SetCookieTest.php]: These tests can potentially fail? Edge case but still... https://blog.frankdejonge.nl/being-in-control-of-time-in-php/ might be a solution, although that's a bit too much for this library...

File: src/SetCookie.php [hunk 3]
```diff
@@ -73,12 +87,63 @@ public static function thatExpires(
...
+     */
+    public static function thatExpiresIn(
```
Comment (hansott) [inline src/SetCookie.php]: Can we add a couple of examples in the readme?

File: src/SetCookie.php (hunk 1 above)
Comment (pine3ree) [reply src/SetCookie.php]: you're totally right, I just missed that! Sorry!!

File: src/SetCookie.php (hunk 3 above)
Comment (pine3ree) [reply src/SetCookie.php]: pleas check if the ones I added seems OK to You!

File: tests/SetCookieTest.php (hunk 2 above)
Comment (pine3ree) [reply tests/SetCookieTest.php]: hi, could you give me an example to see if we can find a non-complicated solution?

Comment (hansott) [reply tests/SetCookieTest.php]: I've been thinking about this (to solve it in a simple way). I prefer to avoid flickering unit tests. I'm not sure yet what the best solution is. I'll come back to you soon! Thanks again for your work.

Comment (pine3ree) [reply tests/SetCookieTest.php]: sure, no problem! :-)
I realized that maybe you were referring to `time()` calls, that can happend in different seconds even if just 1 microsecond apart. In zend-expressive I had to add a double check for min and max values range. The probability of these errors in not 0 and grows bigger with code complexity.

Comment (pine3ree) [reply tests/SetCookieTest.php]: just as an example in 1 of my commits:
https://github.com/zendframework/zend-expressive-session-ext/commit/5ebdb3cb1da0db9aed3efcfdf2e572bcafd381be
but here the situation is simpler as i check timestamps, in a cookie-av expire value there is potentially a difference in the resulting day/month/year for just 1 microseconds...and here we have names...not nice values....

Comment (pine3ree) [reply tests/SetCookieTest.php]: One possible solution is to add an extractor method in the test cases:
`private function extractExpireTimestamp(string$headerValue): int` and use that value to compare against min/max timestamp values. That method could also contain assertion about existence and expected string format of the expire cookie-av value.

Comment (pine3ree) [reply tests/SetCookieTest.php]: ...somethig like this:
```php
    private function extractExpireTimestampFrom(string $headerValue): int
    {
        static $expireRegex
            = '/'
            . '(Sun|Mon|Tue||Wed|Thu|Fri|Sat)\,'
            . ' [1-3][0-9]'
            . ' (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)'
            . ' [0-9]{4}'
            . ' [0-2][0-9]\:[0-5][0-9]:[0-5][0-9]'
            . ' [A-Z]{3}'
            . '/';

        $expireString = null;

        $parts = explode(';', $headerValue);
        foreach ($parts as $part) {
            $av = explode('=', $part);
            $avName = strtolower(trim($av[0]));
            if ($avName === 'expires') {
                $expireString = $av[1] ?? null;
                break;
            }
        }

        $this->assertNotNull($expireString);
        $this->assertRegExp($expireRegex, $expireString);

        $expireTimestamp = strtotime($expireString);

        $this->assertNotFalse($expireTimestamp);

        return $expireTimestamp;
    }

```

**Comment by pine3ree** [comment]
@hansott 
I updated the test cases using range-checking, kind regards.

**Comment by pfk84** [comment]
Any news on this one?

**Comment by hansott** [comment]
Hey @pfk84, I didn't accept the patch because I want to do it the right way: injecting a clock dependency (e.g. https://github.com/symfony/clock)

I can accept a patch if it's the right way. With the current API I don't see a great way to inject the clock. I also have an open issue to improve the API: #15 

Right now, 100% of my time is poured in my startup. I don't have much time for open source, sorry about that.

You're free to fork this package and apply a patch like this.
